server/src/test/java/org/elasticsearch/transport/SingleResultDeduplicatorTests.java

```diff
@@ -88,11 +99,22 @@ public void onFailure(Exception e) {
     public void testThreadContextPreservation() {
         final var resources = new Releasable[1];
         try {
+            final var workerRequestHeaderName = "worker-request-header";
+            final var workerResponseHeaderName = "worker-response-header";
+            final var workerResponseHeaderValueCounter = new AtomicInteger();
+            final ArrayList<Integer> allSeenResponseHeaderValues = new ArrayList<>();
```

Comment:
Needs synchronization I think, we're writing to this list from multiple threads.

Reply:
Thanks for catching it.